src/devtools/client/webconsole/components/Output/ConsoleOutput.js

```diff
@@ -69,12 +72,55 @@ class ConsoleOutput extends Component {
     };
   }
 
+  constructor(props) {
+    super(props);
+    this.maybeScrollToBottom = this.maybeScrollToBottom.bind(this);
+  }
+
   componentDidMount() {
     if (this.props.visibleMessages.length > 0) {
       scrollToBottom(this.outputNode);
     }
   }
 
+  componentDidUpdate(prevProps) {
+    const messagesDelta = this.props.messages.size - prevProps.messages.size;
+
+    // When evaluation results are added, scroll to them.
+    this.shouldScrollMessageId = null;
+    this.shouldScrollMessageNode = null;
```

Comment:
could these be local variables to the function?

Reply:
yes, something i should've changed after removing gtoolbox